pkgs/development/compilers/llvm/12/llvm/default.nix

```
@@ -54,7 +54,8 @@ in stdenv.mkDerivation (rec {
   buildInputs = [ libxml2 libffi ]
     ++ optional enablePFM libpfm; # exegesis
 
-  propagatedBuildInputs = [ ncurses zlib ];
+  propagatedBuildInputs = optionals (stdenv.buildPlatform == stdenv.hostPlatform) [ ncurses ]
```

```suggestion
  propagatedBuildInputs = optionals (stdenv.buildPlatform.isCompatible stdenv.hostPlatform) [ ncurses ]
```

Haven't checked, but I think we can keep ncurses for sets like pkgsStatic.llvm_12 and pkgsCross.gnu32 where we can still run the host's llvm-config on the build system.

---

I don't think this would work, because it doesn't currently build for pkgsStatic:

```
/nix/store/ckymfxng8g4xpnix8apbvyx03xdg9vdb-x86_64-unknown-linux-musl-binutils-2.35.1/bin/x86_64-unknown-linux-musl-ld: /nix/store/l05ny4v41irzq2ns3qk9615h4603pmbr-x86_64-unknown-linux-musl-stage-final-gcc-10.3.0/lib/gcc/x86_64-unknown-linux-musl/10.3.0/crtbeginT.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object
/nix/store/ckymfxng8g4xpnix8apbvyx03xdg9vdb-x86_64-unknown-linux-musl-binutils-2.35.1/bin/x86_64-unknown-linux-musl-ld: /nix/store/l05ny4v41irzq2ns3qk9615h4603pmbr-x86_64-unknown-linux-musl-stage-final-gcc-10.3.0/lib/gcc/x86_64-unknown-linux-musl/10.3.0/crtend.o: relocation R_X86_64_32 against `.ctors' can not be used when making a shared object; recompile with -fPIC
/nix/store/ckymfxng8g4xpnix8apbvyx03xdg9vdb-x86_64-unknown-linux-musl-binutils-2.35.1/bin/x86_64-unknown-linux-musl-ld: /nix/store/acv2cigbl2s428g1cd5zgs0v6hi34wgh-ncurses-6.2-x86_64-unknown-linux-musl/lib/libtinfo.a(free_ttype.o): warning: relocation against `_nc_head' in read-only section `.text'
collect2: error: ld returned 1 exit status
```

---

Ah ok, nvm then